content-repo/gen_top_contrib.py

```diff
@@ -0,0 +1,251 @@
+            pr_bodies.append(pr_body)
+            inner_prs.append(inner_pr[0])
+
+    return pr_bodies, inner_prs
+
+
+def github_pagination_prs(url: str, params: dict, res) -> list:
+    """
+    Paginate through all the pages in Github according to search query
+    Args:
+        url (str): the request url
+        params (dict): params for the request
+        res: the response from the http_request
+
+    Returns: prs (dict)
+    """
+    prs = []
+    if link := res.headers.get('Link'):
+        links = link.split(',')
+        for link in links:
+            last_page = link[link.find("<")+1:link.find(">")][-1]
+        while params['page'] <= int(last_page):
+            params['page'] = params['page'] + 1
+            response = requests.request('GET', url, params=params, headers=HEADERS, verify=False)
+            next_page_prs = response.json().get('items', [])
+            prs.extend(next_page_prs)
+
+    return prs
+
+
+def get_contractors_prs() -> list:
+    """
+    Get all contractors pr numbers and users we want to ignore in the calculation.
+    Returns: The inner pr numbers list of the contractors and the users we want to ignore.
+    """
+    url = URL + '/search/issues'
+    query = 'type:pr state:closed org:demisto repo:content is:merged base:master ' \
+            'head:contrib/crest head:contrib/qmasters head:contrib/mchasepan head:contrib/roybi72'
+    params = {
+        'q': query,
+        'per_page': 100,
+        'page': 1
+    }
+    res = requests.request('GET', url, headers=HEADERS, params=params, verify=False)
+    res.raise_for_status()
+    prs = res.json().get('items', [])
+
+    next_pages_prs = github_pagination_prs(url, params, res)
+    prs.extend(next_pages_prs)
+
+    _, inner_prs = get_external_prs(prs)
+
+    return inner_prs
+
+
+def get_contrib_prs():
+    """
+    Get the contributors prs.
+    Returns: The list of inner PRs and a list of the pr bodies.
+    """
+    url = URL + '/search/issues'
+    query = 'type:pr state:closed org:demisto repo:content is:merged base:master head:contrib/'
+    params = {
+        'q': query,
+        'per_page': 100,
+        'page': 1
+    }
+    res = requests.request('GET', url, headers=HEADERS, params=params, verify=False)
+    res.raise_for_status()
+    prs = res.json().get('items', [])
+
+    # Get all the PRs from all pages
+    next_pages_prs = github_pagination_prs(url, params, res)
+    prs.extend(next_pages_prs)
+
+    pr_bodies, inner_prs = get_external_prs(prs)
+    contractors_prs = get_contractors_prs()
+    for pr in inner_prs:
+        if pr in contractors_prs:
+            inner_prs.remove(pr)
+
+    return pr_bodies, inner_prs
+
+
+def get_github_user(user_name: str):
+    """
+    Get the github user.
+    Args:
+        user_name (str): the github username.
+
+    Returns: The user avatar and its profile.
+
+    """
+    url = f'{URL}/users/{user_name}'
+    res = requests.request('GET', url, headers=HEADERS, verify=False)
```

Comment:
missing `raise_for_status`

Reply:
I don't want to use here raise for status. 
I have a case where user was not found, and getting 404.
I'm handling this with an `if` statement in `get_contributors_users` command.
Don't want this function to fail